pkg/asset/installconfig/baremetal/baremetal.go

```diff
@@ -0,0 +1,142 @@
+// Package baremetal collects bare metal specific configuration.
+package baremetal
+
+import (
+	"fmt"
+	"gopkg.in/AlecAivazis/survey.v1"
+
+	"github.com/openshift/installer/pkg/types/baremetal"
+	baremetaldefaults "github.com/openshift/installer/pkg/types/baremetal/defaults"
+	"github.com/openshift/installer/pkg/validate"
+)
+
+// Platform collects bare metal specific configuration.
+func Platform() (*baremetal.Platform, error) {
+	var externalBridge, provisioningBridge, provisioningNetworkInterface, apiVIP, ingressVIP string
+	var hosts []*baremetal.Host
+
+	err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "External bridge",
+				Help:    "External bridge is used for external communication.",
+				Default: baremetaldefaults.ExternalBridge,
+			},
+		},
+	}, &externalBridge)
+	if err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Provisioning bridge",
+				Help:    "Provisioning bridge is used to provision machines.",
+				Default: baremetaldefaults.ProvisioningBridge,
+			},
+		},
+	}, &provisioningBridge)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Provisioning Network Interface",
+				Help:    "ProvisioningNetworkInterface is the name of the network interface on a control plane host connected to the provisioning network.",
+			},
+			Validate: survey.Required,
+		},
+	}, &provisioningNetworkInterface); err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "API VIP",
+				Help:    "The VIP to be used for internal API communication. If blank, the address will be looked up from DNS.",
+				Default: baremetaldefaults.APIVIP,
+			},
+			Validate: apiVIPValidator,
+		},
+	}, &apiVIP)
+	if err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Ingress VIP",
+				Help:    "The VIP to be used for ingress. If blank, the address will be looked up from DNS.",
+				Default: baremetaldefaults.IngressVIP,
+			},
+			Validate: ingressVIPValidator,
+		},
+	}, &ingressVIP)
+	if err != nil {
+		return nil, err
+	}
+
+	// Keep prompting for hosts
+	for {
+		var hostRole string
+		survey.AskOne(&survey.Select{
+			Message: "Add a Host:",
+			Options: []string{"master", "worker"},
+		}, &hostRole, nil)
+
+		var host *baremetal.Host
+		host, err = Host()
+		if err != nil {
+			fmt.Printf("Invalid host - please try again")
+			continue
+		}
+		hosts = append(hosts, host)
+
+		more := false
+		survey.AskOne(&survey.Confirm{
+			Message: "Add another?",
+		}, &more, nil)
+		if !more {
+			break
+		}
+	}
+
+	return &baremetal.Platform{
```

Comment:
I think we'll need to adjust the survey to cover the whole platform. We should ask questions to cover all the fields in: https://github.com/openshift/installer/blob/master/pkg/types/baremetal/platform.go#L27

Reply:
I've added most of the platform in the survey. PTAL. Thanks.